src/main/scala/performanceanalysis/ParsedLine.scala

```diff
@@ -0,0 +1,66 @@
+package performanceanalysis
+
+import java.time.LocalDateTime
+import java.time.format.{DateTimeFormatter, DateTimeFormatterBuilder, ResolverStyle}
+import java.time.temporal.ChronoField
+
+import scala.util.matching.Regex
+import scala.util.matching.Regex.Match
+
+object LineParser {
+
+  def apply(regex: Regex): LineParser = new LineParser(regex)
+
+  def apply(string: String): LineParser = apply(string.r)
+
+}
+
+class LineParser(regex: Regex) {
+
+  def parse(line: String): ParsedLine = new ParsedLine(line, regex)
+
+}
+
+object ParsedLine {
+
+  val dtRegex = """(\d{1,2})[ -/.](\d{1,2})[ -/.](\d{4})[ T]( \d|\d{2})[:.](\d{2})[.:](\d{2})(\.\d{0,9})?""".r
+
+  def dateParser(iYear: Int, iMonth: Int, iDay: Int): String => Option[LocalDateTime] = {
+    (s: String) => try {
+      val dtr: Option[Match] = dtRegex.findFirstMatchIn(s)
+      dtr.map { m =>
+        val year = m.group(iYear).toInt
+        val month = m.group(iMonth).toInt
+        val day = m.group(iDay).toInt
+        val hour = m.group(4).toInt
+        val min = m.group(5).toInt
+        val sec = m.group(6).toInt
+        val nano = m.group(7) match {
+          case null => 0
+          case "." => 0
+          case s => (s.toDouble * 1000000000).toInt
+        }
+        LocalDateTime.of(year, month, day, hour, min, sec, nano)
+      }
+    } catch {
+      case e: Exception => None
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fatal exception is being caught](https://www.codacy.com/app/scala-academy/performance-analysis/file/2589632303/issues/source?bid=3220915&fileBranchId=3304352#l46)